tests/cpp/build_all_doc_examples.sh

```diff
@@ -5,10 +5,31 @@ script_path=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
 cd "$script_path/../.."
 set -x
 
+
+WERROR=false
+
+while test $# -gt 0; do
+  case "$1" in
+    --werror)
+      shift
+      WERROR=true
+      ;;
+
+    *)
```

Comment:
it would be good to error on unknown options here instead of swallowing them silently (e.g. if the user uses `-werror` or `--Werror`)

Reply:
did so now as well for our publish_crates script (where I copied argument parsing from ;)